### DISCUSSION TITLE
Dev server gracefully quits after short period of time

### DISCUSSION THREAD

**Discussion by dericcain** [vercel/next.js#69922] [Help]
### Summary

A project I contribute to at work is running into some issues with the dev server stopping after a short period of time - usually between 2-10 minutes. Here is some info:

- NextJS v14.2.6
- Yarn v1.22.22
- We are using both the App Router and Page Router simultaneously and we have `/api` routes in the `app` folder and also in the `pages` folder.
- This usually seems to happen after a page navigation, but not always.
- Here is a screenshot of what it looks like when the server quits:
![deric@GCWYQ92THP~Codepartnershipt 2024-09-10 at 9 14 00 AM](https://github.com/user-attachments/assets/5459276b-245b-462a-8a2e-50a7572abaf5)

The main problem I am having is determining why it is stopping and why there isn't an error. I ran a CPU Profile but nothing jumped out (I am not the best at reading those tbh so someone else may find something I couldn't).

**Comment by Juneezee** [comment]
That is really weird. I did some Googling and found some posts:

1. https://stackoverflow.com/q/74074203
2. https://stackoverflow.com/q/75681159
3. https://stackoverflow.com/q/75726370
4. https://www.reddit.com/r/nextjs/comments/11s2lj0/my_server_automatically_closes_after_some_3_to_5

It seems like the issue could be related to Node.js installation. Could you check if you are using Node.js 18 and above, and also installed the correct architecture (e.g. 64-bit instead of 32-bit)?
